.github/workflows/deploy-standalone-plugins.yml

```diff
@@ -1,99 +1,154 @@
-
-            # The "dry-run" parameter is included here to set the deployment mode.
-            # When running the manual (workflow_dispatch) workflow, this value will be set from manual input type.
-            echo "matrix=$(jq -c '{include: ([.modules | to_entries[] | {name:.key, slug: .value.slug, version: .value.version, directory: "build", "dry-run": false}] + ([.plugins[] | {name:. , slug:. , directory: "plugins", "dry-run": false}]))}' plugins.json)" >> $GITHUB_OUTPUT
           fi
+
+          # Set the matrix.
+          echo "::notice::Deploying the following plugins: $PLUGINS"
+          echo "plugins=$(echo $PLUGINS | jq -c .)" >> $GITHUB_OUTPUT
+
   deploy:
-    name: Deploy Plugin
-    needs: release
+    name: "Deploy Plugin: ${{ matrix.plugin }}"
+    needs: pre-run
     runs-on: ubuntu-latest
+    permissions:
+      actions: write
+      deployments: write
     strategy:
-      matrix: ${{ fromJSON(needs.release.outputs.matrix) }}
+      matrix:
+        plugin: ${{ fromJSON(needs.pre-run.outputs.matrix) }}
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
+      - name: Checkout
+        uses: actions/checkout@v4
+
       - name: Setup Node.js (.nvmrc)
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version-file: '.nvmrc'
           cache: npm
+
       - name: Install npm dependencies
         run: npm ci
-      - name: Building standalone plugins
-        if: ${{ matrix.directory != 'plugins' }}
-        run: npm run build-plugins
-      - name: Set version
-        id: set_version
+
+      - name: Build plugin
+        run: npm run build:plugin:${{ matrix.plugin }}
+
+      - name: Set plugin version
+        working-directory: ./build
         run: |
-          if [ -z "${{ matrix.version }}" ]; then
-            echo "version=$(node ./bin/plugin/cli.js get-plugin-version --slug=${{ matrix.slug }})" >> $GITHUB_OUTPUT
-          else
-            echo "version=${{ matrix.version }})" >> $GITHUB_OUTPUT
-          fi
-      - name: Deploy Standalone Plugin - ${{ matrix.slug }}
+          echo "PLUGIN_VERSION=$(jq -r '."${{ matrix.plugin }}"' manifest.json)" >> $GITHUB_ENV
+
+      - name: Create zip file
+        run: |
+          mkdir -p ./build/dist
+          cd ./build/${{ matrix.plugin }}
+          zip -r ../dist/${{ matrix.plugin }}.zip .
+
+      - name: Generate checksum
+        working-directory: ./build/dist
+        run: |
+          mkdir -p $RUNNER_TEMP/plugin-checksums
+          find . -type f -print0 | sort -z | xargs -r0 shasum -a 256 -b | sed 's# \*\./# *#' > $RUNNER_TEMP/plugin-checksums/checksums.txt
+          shasum -a 256 -U -c $RUNNER_TEMP/plugin-checksums/checksums.txt
+          cat $RUNNER_TEMP/plugin-checksums/checksums.txt | while read sum file; do echo "$sum $file" > ${file#\*}.sha256; done
```

Comment:
Do we really need to create checksums? A bit overkill IMO. I doubt anyone would ever look at these.

If someone wants checksum verification, WordPress.org already provides that information anyway through a nice API.

Reply:
It's not a "must use," but providing a means to verify the integrity of your distributed assets is never bad.